### PR TITLE
Added readme generation

### PR DESCRIPTION
**Pull request by vzablock**
## Description
- Issue(s): https://github.com/bitmovin/dashboard/issues/662
- Work: Added readme generation from yaml files

### CODE REVIEW

File: yamlToReadmeConverter.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            parsedTags = parsedTags + `\r\n  - ${tag}`;
+        }
+    });
+    return parsedTags;
+}
+
+const createReadme = (filePath, index) => {
+    YAML.load(filePath, (result) => {
+        const readmePath = folderPaths[index] + '/README.md';
+        const content = `# Bitmovin Demo:\r\n${result.title}\r\n\r\n## Demo Description:\r\n${result.description}\r\n\r\n### Detailed Demo Description:\r\n${result.long_description}\r\n\r\n### Tags:\r\n${parseTags(result.tags)}`;
```
Comment (lukaskroepfl) [inline yamlToReadmeConverter.js]: could we extract this into a template yaml file that is loaded and has placeholders in it?

File: yamlToReadmeConverter.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+const checkAndCreateReadmes = async () => {
+
+await getDirectories();
+
+console.log('Generated all readmes!');
+filePaths.forEach((filePath, index) => createReadme(filePath, index));
+}
+
+checkAndCreateReadmes();
```
Comment (lukaskroepfl) [inline yamlToReadmeConverter.js]: newline missing

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+language: node_js
+node_js: 'lts/*'
+
+install: yarn install
+
+script: node yamlToReadmeConverter.js
```
Comment (lukaskroepfl) [inline .travis.yml]: add newline

File: player/vr-360/info.yaml [hunk 4]
```diff
@@ -1,20 +1,20 @@
-title: VR & 360°
-description: This demo shows the Bitmovin Player displaying a 360° video with VR
-long_description: The Bitmovin Player was the first player on the market to achieve complete cross browser compatibility, and is still leading the way in features and performance.
+title: VR & 360°
```
Comment (lukaskroepfl) [inline player/vr-360/info.yaml]: pls remove the tabs

File: yamlToReadmeConverter.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
+const fs = require('fs');
+const path = require('path');
+const YAML = require('yamljs');
+
+const folderPaths = [];
+const filePaths = [];
+const ignoredFolders = ['.git', 'node_modules', '.travis', 'DS_Store'];
```
Comment (fabiandev) [inline yamlToReadmeConverter.js]: Isn't `.DS_Store` prefixed with a `.`?

File: yamlToReadmeConverter.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            parsedTags = parsedTags + `\r\n  - ${tag}`;
+        }
+    });
+    return parsedTags;
+};
+
+const createReadme = (filePath, index) => {
+    YAML.load(filePath, (result) => {
+        const mapObj = {
+            '{{title}}': result.title,
```
Comment (fabiandev) [inline yamlToReadmeConverter.js]: Title and description are not distinguished at all in the readme, which looks weird in general and especially for demos where they are identical, e.g. for the caption styling [README.md](https://github.com/bitmovin/demos/blob/ed16510027966a4f1b1f48dd9805aa3b39a543f7/player/caption-styling/README.md). The same applies to the long description

File: yamlToReadmeConverter.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            process.exit(1);
+        }
+    })
+};
+
+const createReadmesPromise = new Promise(() => {
+    return checkAndCreateReadmes();
+})
+
+Promise.resolve(createReadmesPromise).then(() => {
```
Comment (fabiandev) [inline yamlToReadmeConverter.js]: Same here.

File: yamlToReadmeConverter.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        else {
+            console.log(folderPath);
+            console.error(`Readme file does not exist in folder: ${showPath(folderPath, '/')}!`);
+            process.exit(1);
+        }
+    })
+};
+
+const createReadmesPromise = new Promise(() => {
```
Comment (fabiandev) [inline yamlToReadmeConverter.js]: Do we actually need a promise here?

File: yamlToReadmeConverter.js [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+const checkAndCreateReadmes = () => {
+    searchDirectories();
+
+    filePaths.forEach((filePath, index) => createReadme(filePath, index));
+    console.log('All readmes present!');
+};
+
+const checkReadmes = () => {
+    folderPaths.forEach((folderPath) => {
+        if (fs.statSync(folderPath).isDirectory() && fs.existsSync(path.join(folderPath, 'README.md'))) {
```
[the remark this answers is not in the document]
Comment (vzablock) [reply yamlToReadmeConverter.js]: Checking the readmes is done at the end, the YAML files are always recreated.

File: package.json [hunk 10]
```diff
@@ -1,4 +1,7 @@
 {
+  "scripts": {
+    "start": "node ./scripts/yamlToReadmeConverter.js"
```
Comment (fabiandev) [inline package.json]: `start` is probably not the best name for the readme generation. Something like `generate-readmes` and `check-readmes`?

File: scripts/checkYamlAndReadme.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+        folderWalk(folderPath);
+    }
+};
+
+const checkFiles = () => {
+    searchDirectories();
+    console.log('All files present!');
+}
+
+checkFiles();
```
Comment (fabiandev) [inline scripts/checkYamlAndReadme.js]: Missing new line.

Comment (vzablock) [reply scripts/checkYamlAndReadme.js]: Added new line

**Pull request merged**